src/mono/wasm/features.md

```diff
@@ -2,36 +2,311 @@
+
+You can bundle the `dotnet.js` ES6 module with the rest of your JavaScript application.
+
+The other assets and JS modules of the dotnet are loaded via dynamic `import` or via `fetch` APIs. They are not ready to be bundled in Net8.
+We consider that the dotnet application is usually large as is and giving the browser chance to start compiling it in parallel with other downloads is better.
+
+We would like to [hear from the community](https://github.com/dotnet/runtime/issues/86162) more about the use-cases when it would be benefitial.
+
+# Resources consumed on the target device
+
+dotnet is complex and large application, it consists of
+- dotnet runtime, including garbage collector, IL interpreter and browser specific JIT
+- dotnet base class library
+- emulation layer which is bringing missing features of the OS, which the browser doesn't provide. Like timezone database or ICU.
+- integration with the browser JavaScript APIs, for example HTTP and WebSocket client
+- application code
+
+All of the mentioned code and data need to be downloaded and loaded into the browser memory during the dotnet startup sequence.
+Browser itself will run JIT compilation of the WASM and JS code, which consumes memory and CPU cycles too.
+
+## Mobile phones
+
+Recent mobile phones distribute their browser as an application that can be upgraded separately from the operating system.
+
+Note that all browsers on iOS and iPadOS are required to use the Safari browser engine, so their level of support for WASM features depends on the version of Safari installed on the device.
+
+Mobile browsers typically have strict limits on the amount of memory they can use, and many users are on slow internet connections. A WebAssembly application that works well on desktop PCs may take minutes to download or run out of memory before it is able to start.
+
+## Shell environments - NodeJS & V8
 We pass most of the unit tests with NodeJS v 14 but it's not fully supported target platform. We would like to hear about community use-cases.
-We also use v8 engine version 11 or higher to run some of the tests. The engine is lacking many APIs and features.
\ No newline at end of file
+
+We also use the d8 command-line shell, version 11 or higher, to run some of the tests. This shell lacks most browser APIs and features.
+
+## Trade-offs compared to native JavaScript applications
+
+There is trade-off between download size, application performance and complexity of application maintanance.
+If your business logic is very complex, changes often or you already have existing C# codebase and skillset, running the same code dotnet on wasm is probably the right choice.
+If your application is simple and you have JavaScript skills on your team, it may be better if you re/write your logic in Web native technologies like HTML/JavaScript or typescript/webpack stack.
```

Comment:
```suggestion
- If your business logic is very complex, changes often or you already have existing C# codebase and skillset, then running the same code with dotnet on wasm is probably the right choice.
- If your application is simple and you have JavaScript skills on your team, it may be better if you re/write your logic in Web native technologies like HTML/JavaScript or typescript/webpack stack.
```